# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTCAudio.cpp
# Commit: 3aef6c404e07
# Full Hash: 3aef6c404e07742214a5e3c1ccbcbd2ad1eee959
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2022-01-28 04:41:16
# Regressor Bug: 1742655
# File Overlap Count: 2
# Description:
#   Bug 1742655 - Rename AudioInputTrack to AudioProcessingTrack r=pehrsons
#   
#   Depends on D134628
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134629
# ==============================================================================

diff -r b004b579d1ce -r 3aef6c404e07 dom/media/webrtc/MediaEngineWebRTCAudio.cpp
--- a/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Thu Jan 27 22:29:35 2022 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Thu Jan 27 22:29:35 2022 +0000
@@ -1169,27 +1169,27 @@
   mChunksInPacketizer.clear();
 }
 
-void AudioInputTrack::Destroy() {
+void AudioProcessingTrack::Destroy() {
   MOZ_ASSERT(NS_IsMainThread());
   DisconnectDeviceInput();
 
   MediaTrack::Destroy();
 }
 
-void AudioInputTrack::SetInputProcessing(
+void AudioProcessingTrack::SetInputProcessing(
     RefPtr<AudioInputProcessing> aInputProcessing) {
   class Message : public ControlMessage {
-    const RefPtr<AudioInputTrack> mTrack;
+    const RefPtr<AudioProcessingTrack> mTrack;
     const RefPtr<AudioInputProcessing> mProcessing;
 
    public:
-    Message(RefPtr<AudioInputTrack> aTrack,
+    Message(RefPtr<AudioProcessingTrack> aTrack,
             RefPtr<AudioInputProcessing> aProcessing)
         : ControlMessage(aTrack),
           mTrack(std::move(aTrack)),
           mProcessing(std::move(aProcessing)) {}
     void Run() override {
-      TRACE("AudioInputTrack::SetInputProcessingImpl");
+      TRACE("AudioProcessingTrack::SetInputProcessingImpl");
       mTrack->SetInputProcessingImpl(mProcessing);
     }
   };
@@ -1201,27 +1201,28 @@
       MakeUnique<Message>(std::move(this), std::move(aInputProcessing)));
 }
 
-AudioInputTrack* AudioInputTrack::Create(MediaTrackGraph* aGraph) {
+AudioProcessingTrack* AudioProcessingTrack::Create(MediaTrackGraph* aGraph) {
   MOZ_ASSERT(NS_IsMainThread());
-  AudioInputTrack* track = new AudioInputTrack(aGraph->GraphRate());
+  AudioProcessingTrack* track = new AudioProcessingTrack(aGraph->GraphRate());
   aGraph->AddTrack(track);
   return track;
 }
 
-void AudioInputTrack::DestroyImpl() {
+void AudioProcessingTrack::DestroyImpl() {
   ProcessedMediaTrack::DestroyImpl();
   if (mInputProcessing) {
     mInputProcessing->End();
   }
 }
 
-void AudioInputTrack::ProcessInput(GraphTime aFrom, GraphTime aTo,
-                                   uint32_t aFlags) {
-  TRACE_COMMENT("AudioInputTrack::ProcessInput", "AudioInputTrack %p", this);
+void AudioProcessingTrack::ProcessInput(GraphTime aFrom, GraphTime aTo,
+                                        uint32_t aFlags) {
+  TRACE_COMMENT("AudioProcessingTrack::ProcessInput", "AudioProcessingTrack %p",
+                this);
   MOZ_ASSERT(mInputProcessing);
 
   LOG_FRAME(
-      "(Graph %p, Driver %p) AudioInputTrack %p ProcessInput from %" PRId64
+      "(Graph %p, Driver %p) AudioProcessingTrack %p ProcessInput from %" PRId64
       " to %" PRId64 ", needs %" PRId64 " frames",
       mGraph, mGraph->CurrentDriver(), this, aFrom, aTo, aTo - aFrom);
 
@@ -1233,7 +1234,7 @@
     MOZ_ASSERT(TrackTimeToGraphTime(GetEnd()) == aFrom);
     if (mInputs.IsEmpty()) {
       GetData<AudioSegment>()->AppendNullData(aTo - aFrom);
-      LOG_FRAME("(Graph %p, Driver %p) AudioInputTrack %p Filling %" PRId64
+      LOG_FRAME("(Graph %p, Driver %p) AudioProcessingTrack %p Filling %" PRId64
                 " frames of null data (no input source)",
                 mGraph, mGraph->CurrentDriver(), this, aTo - aFrom);
     } else {
@@ -1251,9 +1252,10 @@
   }
 }
 
-void AudioInputTrack::GetInputSourceData(AudioSegment& aOutput,
-                                         const MediaInputPort* aPort,
-                                         GraphTime aFrom, GraphTime aTo) const {
+void AudioProcessingTrack::GetInputSourceData(AudioSegment& aOutput,
+                                              const MediaInputPort* aPort,
+                                              GraphTime aFrom,
+                                              GraphTime aTo) const {
   MOZ_ASSERT(mGraph->OnGraphThread());
   MOZ_ASSERT(aOutput.IsEmpty());
 
@@ -1278,18 +1280,18 @@
 
     if (inputEnded) {
       aOutput.AppendNullData(ticks);
-      LOG_FRAME("(Graph %p, Driver %p) AudioInputTrack %p Getting %" PRId64
+      LOG_FRAME("(Graph %p, Driver %p) AudioProcessingTrack %p Getting %" PRId64
                 " ticks of null data from input port source (ended input)",
                 mGraph, mGraph->CurrentDriver(), this, ticks);
     } else if (interval.mInputIsBlocked) {
       aOutput.AppendNullData(ticks);
-      LOG_FRAME("(Graph %p, Driver %p) AudioInputTrack %p Getting %" PRId64
+      LOG_FRAME("(Graph %p, Driver %p) AudioProcessingTrack %p Getting %" PRId64
                 " ticks of null data from input port source (blocked input)",
                 mGraph, mGraph->CurrentDriver(), this, ticks);
     } else if (source->IsSuspended()) {
       aOutput.AppendNullData(ticks);
       LOG_FRAME(
-          "(Graph %p, Driver %p) AudioInputTrack %p Getting %" PRId64
+          "(Graph %p, Driver %p) AudioProcessingTrack %p Getting %" PRId64
           " ticks of null data from input port source (source is suspended)",
           mGraph, mGraph->CurrentDriver(), this, ticks);
     } else {
@@ -1305,9 +1307,10 @@
   }
 }
 
-void AudioInputTrack::NotifyOutputData(MediaTrackGraphImpl* aGraph,
-                                       AudioDataValue* aBuffer, size_t aFrames,
-                                       TrackRate aRate, uint32_t aChannels) {
+void AudioProcessingTrack::NotifyOutputData(MediaTrackGraphImpl* aGraph,
+                                            AudioDataValue* aBuffer,
+                                            size_t aFrames, TrackRate aRate,
+                                            uint32_t aChannels) {
   MOZ_ASSERT(mGraph == aGraph, "Cannot feed audio output to another graph");
   MOZ_ASSERT(mGraph->OnGraphThread());
   if (mInputProcessing) {
@@ -1316,13 +1319,13 @@
   }
 }
 
-void AudioInputTrack::SetInputProcessingImpl(
+void AudioProcessingTrack::SetInputProcessingImpl(
     RefPtr<AudioInputProcessing> aInputProcessing) {
   MOZ_ASSERT(GraphImpl()->OnGraphThread());
   mInputProcessing = std::move(aInputProcessing);
 }
 
-nsresult AudioInputTrack::ConnectDeviceInput(
+nsresult AudioProcessingTrack::ConnectDeviceInput(
     CubebUtils::AudioDeviceID aId, AudioDataListener* aListener,
     const PrincipalHandle& aPrincipal) {
   MOZ_ASSERT(NS_IsMainThread());
@@ -1346,7 +1349,7 @@
   return NS_OK;
 }
 
-void AudioInputTrack::DisconnectDeviceInput() {
+void AudioProcessingTrack::DisconnectDeviceInput() {
   MOZ_ASSERT(NS_IsMainThread());
   MOZ_ASSERT(GraphImpl());
   if (!mInputListener) {
@@ -1363,7 +1366,7 @@
   mDeviceId = Nothing();
 }
 
-Maybe<CubebUtils::AudioDeviceID> AudioInputTrack::DeviceId() const {
+Maybe<CubebUtils::AudioDeviceID> AudioProcessingTrack::DeviceId() const {
   MOZ_ASSERT(NS_IsMainThread());
   return mDeviceId;
 }